async-nats/src/client.rs

```diff
@@ -574,3 +579,48 @@ impl Request {
         self
     }
 }
+
+#[derive(Error, Debug)]
+#[error("failed to send subscribe")]
+pub struct SubscribeError(#[from] mpsc::error::SendError<Command>);
+
+#[derive(Debug, Error)]
+pub enum RequestError {
+    #[error("request timed out")]
+    TimedOut,
+    #[error("no responders")]
+    NoResponders,
```

Comment:
I'm still in favor of having something like Status(StatusCode) for status errors.

Reply:
That could work but it will not, becase there are 5 or more different errors with status 408 and 409.